### PR TITLE
Add SkyPilot and AWS configuration files

### PR DESCRIPTION
**Pull request by daveey**
### TL;DR

Added AWS and SkyPilot configuration files to streamline cloud deployment for training jobs.

### What changed?

- Added AWS configuration files with profile settings for different access levels
- Created SkyPilot configuration files and launch scripts for running training jobs in the cloud
- Added documentation for SkyPilot setup and usage
- Updated Brewfile to include FUSE for mounting S3 buckets
- Added Cursor editor configuration files for AWS and SkyPilot directories
- Added new words to the workspace dictionary

### How to test?

1. Launch a training job using the new SkyPilot script:
```bash
./devops/skypilot/launch.sh train my_experiment_001
```

2. Monitor job status:
```bash
sky jobs list
sky jobs logs <JOB_ID>
```

3. Test AWS configuration by using the profiles:
```bash
aws --profile softmax-db s3 ls
```

### Why make this change?

This change provides a standardized way to launch and manage training jobs on AWS cloud infrastructure using SkyPilot. It simplifies the deployment process, ensures consistent environment setup, and enables efficient resource utilization through spot instances. The configuration files and documentation make it easier for team members to run experiments in the cloud without having to manually configure each job.

### CODE REVIEW

**Comment by daveey** [comment]
* **#362** <a href="https://app.graphite.dev/github/pr/Metta-AI/metta/362?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Metta-AI/metta/362?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: devops/skypilot/config/setup.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# Install SkyPilot
+pip install skypilot
```
Comment (berekuk) [inline devops/skypilot/config/setup.sh:4]: is the assumed order `install.sh` -> `setup.sh` -> `launch.sh`? can we merge "install" and "setup"?

File: devops/skypilot/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+## Examples
+
+1. Launch a training run with default parameters:
+```bash
+./launch.sh train my_experiment_001
+```
+
+2. Launch a training run with specific arguments:
+```bash
+./launch.sh train my_experiment_002 learning_rate=0.001 batch_size=32
```
Comment (berekuk) [inline devops/skypilot/README.md:32]: shouldn't this be `trainer.learning_rate=0.001 trainer.batch_size=32`?

File: devops/aws/.cursor [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    ]
+  },
+  "tasks": {
+    "enabled": true,
+    "scripts": {
+      "setup": "python setup_sso.py",
+      "cmd": "./cmd.sh"
+    }
+  }
+}
```
Comment (berekuk) [inline devops/aws/.cursor:54]: What is this file? I can't find documentation for it anywhere, it's not `.cursorrules`and not VS Code-like settings

File: devops/aws/aws_config.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+[profile softmax]
+sso_session = softmax-sso
+sso_account_id = 767406518141
+sso_role_name = PowerUserAccess
+region = us-east-1
+
+[profile softmax-admin]
+sso_session = softmax-sso
+sso_account_id = 767406518141
```
Comment (berekuk) [inline devops/aws/aws_config.txt:25]: these use your previous personal account, are we not ready to move yet?

File: devops/skypilot/config/train.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  ~/.netrc: ~/.netrc
+
+  /mnt/s3/softmax-public:
+    source: s3://softmax-public
+    mode: MOUNT
+
+  /workspace/metta/train_dir:
+    name: train-dir
+    store: s3
+    mode: MOUNT
```
Comment (berekuk) [inline devops/skypilot/config/train.yaml:45]: My attempts to run this according to instructions haven't succeeded yet.

`sky api logs` show this:
```
⚙︎ Translating file_mounts with local source paths to SkyPilot Storage...
  Files (listed below)  -> storage: skypilot-filemounts-berekuk-1a6f4f22-8mvhle09:
    /Users/berekuk/.netrc
Could not mount externally created StoreType.S3 store with name 'train-dir'.
```

**Review by berekuk — COMMENTED** [review]
I'm happy about this and getting rid of all custom batch code.

**Comment by daveey** [comment]
Could you approve?

File: devops/skypilot/config/task.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  cloud: aws
+  region: us-east-1
+  instance_type: g6.4xlarge
+  use_spot: true
+  num_nodes: 1
+  accelerators: {A10G: 4}  # 4 GPUs per node
+
+setup: |
+  # Install dependencies
+  pip install -r requirements.txt
```
[the remark this answers is not in the document]
Comment (daveey) [reply devops/skypilot/config/task.yaml:11]: I'm trying to get rid of setup build and be more explicit about reqs

File: devops/aws/.cursor (hunk 3 above)
Comment (daveey) [reply devops/aws/.cursor:54]: i tried to get cursor to configure itself, i guess it did a bad job

File: devops/aws/aws_config.txt (hunk 4 above)
Comment (daveey) [reply devops/aws/aws_config.txt:25]: yeah, we need ECR on the new account, and i haven't pushed that yet

File: devops/skypilot/config/setup.sh (hunk 1 above)
Comment (daveey) [reply devops/skypilot/config/setup.sh:4]: setup was llm generated, i used it as an example but forgot to remove.

File: devops/skypilot/config/train.yaml (hunk 5 above)
Comment (daveey) [reply devops/skypilot/config/train.yaml:45]: are you using AWS_PROFILE=softmax-db-admin when launching?

File: devops/skypilot/install.sh [hunk 7]
```diff
@@ -0,0 +1,12 @@
+#! /bin/zsh -e
+
+# Recommended: use a new conda env to avoid package conflicts.
+# SkyPilot requires 3.7 <= python <= 3.11.
+conda create -y -n sky python=3.10
+conda activate sky
```
[the remark this answers is not in the document]
Comment (daveey) [reply devops/skypilot/install.sh]: just put it into reqs

File: requirements.txt [hunk 8]
```diff
@@ -27,6 +27,7 @@ rich==13.9.4
 ruff==0.11.8
 scipy==1.14.1
 shimmy==1.3.0
+skypilot==0.8.1
```
Comment (berekuk) [inline requirements.txt:30]: looks like `--async` was introduced in skypilot 0.9.1 (last week).

0.9.1 requires the dev version of omegaconf though, so pip install it but errors:

```
  Attempting uninstall: skypilot
    Found existing installation: skypilot 0.8.1
    Uninstalling skypilot-0.8.1:
      Successfully uninstalled skypilot-0.8.1
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
hydra-core 1.3.2 requires omegaconf<2.4,>=2.2, but you have omegaconf 2.4.0.dev3 which is incompatible.
Successfully installed aiofiles-24.1.0 httpcore-1.0.9 httptools-0.6.4 httpx-0.28.1 omegaconf-2.4.0.dev3 python-multipart-0.0.20 skypilot-0.9.1 uvloop-0.21.0 watchfiles-1.0.5 websockets-15.0.1
```

File: devops/skypilot/config/train.yaml (hunk 5 above)
Comment (berekuk) [reply devops/skypilot/config/train.yaml:45]: Yes. (your launch script does it itself; I definitely have `softmax-db-admin` in my config, so it's not the cause.

```
$ AWS_PROFILE=softmax-db-admin aws sts get-caller-identity | cat
{
    "UserId": "AROA3FLH6F56UIVYZPCFV:slava@stem.ai",
    "Account": "767406518141",
    "Arn": "arn:aws:sts::767406518141:assumed-role/AWSReservedSSO_AdministratorAccess_55d4d0c48cdb34bf/slava@stem.ai"
}
```
